snapcraft/internal/meta/snap.py

```diff
@@ -127,6 +128,21 @@ def get_provider_content_directories(self) -> Set[str]:
 
         return provider_dirs
 
+    def _get_required_grade(self) -> str:
+        if self.__required_grade is not None:
+            return self.__required_grade
+
+        if self.base is None:
```

Comment:
is it possible base is None, if project is built without a base?

Reply:
hmm, correct, this can happen for type: kernel, base and snapd now... for apps and gadget now, you would be reexecuting into the old legacy code base for that

Reply:
Hmmmm, type:kernel should be declaring at least build-base. When it includes an initrd, and binaries come from ~= matching build-base.

Reply:
kernel's do not have bases though, the build-base is just to determine the build environment